### PR TITLE
feat(message): add tab completion and mentions

### LINKED ISSUE #58: Autocompletion of usernames in chat + mentioning feature
Implement `@user` syntax with autocomplete and mentioning (if oneself is mentioned in chat).

### PR DESCRIPTION
**Pull request by olastor**
depends on https://github.com/Sirimangalo/meditation-plus-server/pull/139

closes #58

### CODE REVIEW

File: src/app/message/message.component.ts [hunk 1]
```diff
@@ -69,10 +83,29 @@ export class MessageComponent implements OnInit, OnDestroy {
         if (page > 0 && data.length === 0) {
           this.noMorePages = true;
         }
+
+        this.extractUsernames(data);
       }, () => this.loadingPage = false);
   }
 
   /**
+   * Extract and sort all usernames from message data
```
Comment (binarious) [inline src/app/message/message.component.ts]: What if I'd like to mention a person who didn't post a message, yet?

Comment (olastor) [inline src/app/message/message.component.ts]: Yeah, that's not possible with this function. I'll try to add a request route to solve that problem.

**Comment by olastor** [comment]
The implementation is pretty basic. There is also no solution yet for when a user types a string that matches multiple usernames, in this case always the first one (alphabetically) is chosen. Completing let's say the string `'m'` for the set of usernames `usernames = ['mathew', 'matt']`, the best outcome is to get as a result `'mat'` (the string that matches both). But doing this is actually harder than it seems and algorithms behind that have some more complexity and require different kind of data structures (https://en.wikipedia.org/wiki/Radix_tree) to run efficiently. This maybe would be an idea for further improvement, but is for now a bit too complex, I think.

File: src/app/message/mentions.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl, SafeUrl} from '@angular/platform-browser';
+
+/*
+ * Replace any occurence of a '@username' like string with
+ * a link to the profile page of this user
+*/
+@Pipe({name: 'mentions'})
```
Comment (binarious) [inline src/app/message/mentions.pipe.ts]: please add `pure: true`

**Pull request merged**